src/inspector_agent.cc

```diff
@@ -625,6 +605,37 @@ void Agent::RegisterAsyncHook(Isolate* isolate,
                               v8::Local<v8::Function> disable_function) {
   enable_async_hook_function_.Reset(isolate, enable_function);
   disable_async_hook_function_.Reset(isolate, disable_function);
+  if (pending_enable_async_hook_) {
+    pending_enable_async_hook_ = false;
+    EnableAsyncHook(isolate);
+  } else if (pending_disable_async_hook_) {
+    pending_disable_async_hook_ = false;
+    DisableAsyncHook(isolate);
+  }
+}
+
+void Agent::EnableAsyncHook(Isolate* isolate) {
```

Comment:
It would be good to add a `HandleScope` here and in the method below because currently they leak handles into the caller's `HandleScope`.  It's harmless right now but could become an issue in future refactorings.

Reply:
Done.